src/actions/guides/http_and_routing/routing_and_params.cr

```diff
@@ -247,6 +243,35 @@ class Guides::HttpAndRouting::RoutingAndParams < GuideAction
 
     > The `fallback` should always contain a `Lucky::RouteNotFoundError` error. This is to throw a 404 when an asset, or some other file is not found.
 
+    ## Memoization
+
+    As your application gets larger, you may need to write helper methods that run expensive
+    calculations, or queries. Calling these methods multiple times can lead to performance issues.
+    To mitigate these, you can use the `memoize` macro.
+
+    ```crystal
+    class Reports::Show < BrowserAction
+      get "/report" do
+        small_number = calculate_numbers
+        big_number = calculate_numbers + 1000
+        render ShowPage, small_number: small_number, big_number: big_number
+      end
+
+      memoize def calculate_numbers
+        # This is ran only the first time it's called
+        ReportQuery.new.fetch_numbers_for_today(Time.utc)
+      end
+    end
+    ```
+
+    The `memoize` macro can be used anywhere in your application to memoize a method. However,
```

Comment:
This isn't quite right I think. It can be used in Actions because we include Lucky::Memoizable. It can be used elsewhere but you need to include that module first. So I think we should mention that, Other than that this looks great

Reply:
Ah that's true. Maybe we remove it for now, but in the update then we can add examples on using it in other places. Less confusion for now. Plus we may end up updating the macro to support more things.

Reply:
That sounds good to me 👍